### ISSUE TITLE
 * ERROR: Failed to run install_debian_check_services()!!!

### ISSUE THREAD

**Issue by jsplink** [saltstack/salt-bootstrap#366]
```
Setting up python-m2crypto (0.20.1-1+b1) ...
Setting up python-mako (0.7.0-1.1~bpo60+1) ...
Setting up python-pkg-resources (0.6.14-4) ...
Setting up python-yaml (3.09-5) ...
Setting up rsync (3.0.7-2) ...
update-rc.d: using dependency based boot sequencing
Setting up salt-common (2014.1.1+ds-1~bpo60+1) ...
Processing triggers for python-central ...
Setting up salt-minion (2014.1.1+ds-1~bpo60+1) ...
Starting salt minion control daemon: salt-minion.
Processing triggers for python-support ...
 *  INFO: Running install_debian_check_services()
 * ERROR: Failed to run install_debian_check_services()!!!
```

**Comment by jsplink** [comment]
when running the new `. deploy.sh` on a failed (b/c of the pkg authorization error) cloud.map deployed instance. If there was a way to designate the `salt-cloud --deploy-src <new bootstrap location> -m cloud.map` this would be a more controlled issue report. Filed a feature request for this in the salt main repo here... https://github.com/saltstack/salt/issues/11782

**Comment by s0undt3ch** [comment]
Can I get the full debug output from the script when you additionally pass `-D`?

**Comment by jsplink** [comment]
Looks like salt-minion isn't being run prior to something that needs it running

```
[ ... ]
Reading state information...
python-zmq is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 59 not upgraded.
 *  INFO: Running install_debian_6_stable()
Reading package lists...
Building dependency tree...
Reading state information...
salt-minion is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 59 not upgraded.
 *  INFO: Running install_debian_check_services()
 * DEBUG: Checking if service salt-minion is enabled
 * DEBUG: Service salt-minion is NOT enabled
 * ERROR: Failed to run install_debian_check_services()!!!
```

**Comment by s0undt3ch** [comment]
The **full** output, from start to finish?

**Comment by nrgaway** [comment]
I have same issue.  Log output is here: http://pastebin.com/MnqU55G3

The install script also fails to provide 2 dependencies `procps` and `pciutils`

The issue why it is failing is because there is no runlevel set (I'm using a docker) and the install script is using the runlevel as part of the check to see if service is active.  Since runlevel will return only 1 value (which is 'unknown') the awk parse receives nothing.

```
if [ -f /etc/rc$(runlevel | awk '{ print $2 }').d/S*${servicename} ]; then
     echo "Service ${servicename} is enabled"
     exit 0
 else
     echo "Service ${servicename} is NOT enabled"
     exit 1
 fi
```

Here is a hack (set before running script):

```
ln -s /etc/rc2.d/ /etc/rc.d
```

Here is a partial copy of my docker script

```
FROM debian:wheezy
MAINTAINER Jason Mehring <nrgaway@gmail.com>

# Update distribution
ENV DEBIAN_FRONTEND noninteractive
RUN apt-get -q update
RUN apt-get -y dist-upgrade

ADD ./srv /srv
ADD ./srv/salt/salt/files/salt /etc/salt

# Salt install script uses runlevel (which is unknown) to help determine if packages are installed
# This is a hack so installation will complete
RUN ln -s /etc/rc2.d/ /etc/rc.d

# Install salt-minion
RUN apt-get -q update
RUN apt-get install -y wget ca-certificates
RUN wget -O /root/install_salt.sh http://bootstrap.saltstack.org
RUN sh /root/install_salt.sh -D -X -U -p procps -p pciutils

RUN salt-call --local state.highstate -l debug
```

**Comment by s0undt3ch** [comment]
Bootstrapping a docker instance is not supported.

**Comment by s0undt3ch** [comment]
However, you're telling me @nrgaway that if you install those additional packages it works?

**Comment by nrgaway** [comment]
So far so good.  I just got it working an hour ago after tracking down the installation problem and dependency issues.

The docker file I listed above successfully installed salt-minion using the stable build.  I will be testing it over the next few days since I will be using salt for my DockerNAS project.  I figure it will take about two more weeks for a pimped out docker controlled by salt.  

The DockerNAS project will have a master docker that will control all the minion dockers, which for our purposes are going to be mostly plug-ins like Plex Media Server, postfix, nut, etc.  I figure on at least 50 of them running on a system. I'm adding in a web UI to allow installation and simple configuration where the end user can configure the actual application from the master docker (configuration of application stored in pillar so it can be modified via web UI and changes are sent to minion.  Any updates on minion are immediately sent to master.

I will get all dependency issues sorted out within the next few days and let you know what they are (The `git develop` install option fails on `requests` dependency. )

**Comment by s0undt3ch** [comment]
Yes the requests dependency is on the TODO list, see #363 

Thank you for the investigation work you've done so far. I'll help others I'm sure.

**Comment by s0undt3ch** [comment]
@nrgaway can you test with https://raw.githubusercontent.com/s0undt3ch/salt-bootstrap/develop/bootstrap-salt.sh ? This should install your deps and also requests.

**Comment by nrgaway** [comment]
@s0undt3ch, I ran the tests as requested and found two issues.  The first issue was a regression bug that would effect all Debian users.  I have submitted a PR #367 that solves both issues.  With the modifications I made and the additional dependencies you added salt now bootstraps without issue on a Debian docker container.

The first issue was that https://github.com/saltstack/salt-bootstrap/blob/develop/bootstrap-salt.sh#L1371 was completely quoted which would inhibit globbing.

The second issue was related to environments that had an unknown run level. The bootstrap installation relies on the runlevel to confirm salt has been successfully installed.  Normally the `runlevel` program is supposed to use the environment variable `RUNLEVEL` and `PREVLEVEL` if the `/var/run/utmp` is not available.  This is not the case though, at least for docker running Debian.  

To be able to successfully install with a system with an unknown runlevel you only need set the RUNLEVEL environment variable and it will be used ONLY if the runlevel in unknown.

Example:

```
RUNLEVEL=2 install_salt.sh -D -U
```
